### PR TITLE
Enable digest deduplication for use_artifact calls

### PR DESCRIPTION
**Pull request by annirudh**
Description
-----------
Currently in the `use_artifact` flow, artifacts are recommitted if they have the same logical digest instead of reusing an existing artifact. Newer backends support the option to perform a logical dedupe in the `createArtifact` mutation to reuse an existing artifact instead of creating a new one.

Testing
-------
How was this PR tested?

- [x] New unit tests for `use_artifact`
- [x] End-to-end tests of the `use_artifact` flow

Checklist
-------
- Name PR "[WB-NNNN][WB-MMMM] Add support for..." similar to entries in CHANGELOG.md
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

**Comment by borisdayma** [comment]
I love that the default is to merge aliases. I think it's a sensible default!

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -2126,15 +2126,16 @@ def create_artifact(
...
     ):
-        # TODO: Ignore clientID and sequenceClientID if server can't handle it
         _, server_info = self.viewer_server_info()
         max_cli_version = server_info.get("cliVersionInfo", {}).get(
             "max_cli_version", None
         )
         can_handle_client_id = max_cli_version is None or parse_version(
             "0.11.0"
         ) <= parse_version(max_cli_version)
+        can_handle_dedupe = max_cli_version is None or parse_version("0.12.10")
```
Comment (dmitryduev) [inline wandb/sdk/internal/internal_api.py:2138]:
```suggestion
        can_handle_dedupe = max_cli_version is None or parse_version(
            "0.12.10"
        ) <= parse_version(max_cli_version)
```

**Review by dmitryduev — COMMENTED** [review]
LGTM. I would double-check these two tests https://app.circleci.com/pipelines/github/wandb/client/9816/workflows/f8cf503b-383f-4b8e-85a1-334def0c73fe/jobs/106188 -- it's suspicious that two artifacts-related tests fail on an artifacts-related pr ;)

File: wandb/sdk/internal/internal_api.py (hunk 1 above)
Comment (annirudh) [reply wandb/sdk/internal/internal_api.py]: Woah, I didn't know you could suggest directly on GitHub. Sick. Thanks!

**Comment by annirudh** [comment]
> LGTM. I would double-check these two tests https://app.circleci.com/pipelines/github/wandb/client/9816/workflows/f8cf503b-383f-4b8e-85a1-334def0c73fe/jobs/106188 -- it's suspicious that two artifacts-related tests fail on an artifacts-related pr ;)

The new test I added was broken (it didn't call `run.finish()` nor did it pass in the `test_settings`). Should be fixed now, waiting on CI 🤞 .

**Review by dmitryduev — APPROVED** [review]
👍🏻

**Pull request merged**